src/Orleans/Core/GrainClient.cs

```diff
@@ -381,6 +381,15 @@ public static Streams.IStreamProvider GetStreamProvider(string name)
             return RuntimeClient.Current.CurrentStreamProviderManager.GetProvider(name) as Streams.IStreamProvider;
         }
 
+        public delegate void ConnectionToClusterLostHandler(object sender, EventArgs e);
+
+        public static event ConnectionToClusterLostHandler ClusterConnectionLost;
+
+        internal static void NotifyClusterConnectionLost()
+        {
+            ClusterConnectionLost?.Invoke(null, null);
```

Comment:
`sender` should conceptually be the `GrainClient` instance. Except `GrainClient` is still static. Once we make it non-static, we should send a reference to it here. It will especially make sense for the scenarios with multiple clients connecting to different clusters. The question is what to pass before then. As ugly as it looks, I think maybe `null` as a `sender` is fine in these temporary circumstances.

Is there something meaningful we'd pass as `EventArgs`?